tests/parser/LgrTests.cpp

```diff
@@ -114,3 +115,406 @@ SCHEDULE
     const auto& lgr3 = state.getLgrs().getLgr(0);
     BOOST_CHECK_EQUAL( lgr1.NAME() , lgr3.NAME());
 }
+
+
+BOOST_AUTO_TEST_CASE(TestLgrNeighbor) { 
+    const std::string deck_string = R"(
+RUNSPEC
+
+DIMENS
+  3 3 1 /
+
+GRID
+
+CARFIN
+-- NAME I1-I2 J1-J2 K1-K2 NX NY NZ
+'LGR1'  2  2  2  2  1  1  3  3   /
+ENDFIN
+
+CARFIN
+-- NAME I1-I2 J1-J2 K1-K2 NX NY NZ
+'LGR2'  2  2  1  1  1  1  3  3   /
+ENDFIN
+
+
+DX 
+  9*1000 /
+DY
+	9*1000 /
+DZ
+	9*20 /
+
+TOPS
+	9*8325 /
+
+PORO
+  9*0.15 /
+
+PERMX
+  9*1 /
+
+COPY
+  PERMX PERMZ /
+  PERMX PERMY /
+/
+
+EDIT
+
+OIL
+GAS
+
+TITLE
+The title
+
+START
+16 JUN 1988 /
+
+PROPS
+
+REGIONS
+
+SOLUTION
+
+SCHEDULE
+)";
+\
+    Opm::Parser parser;
+    Opm::Deck deck = parser.parseString(deck_string);
+    Opm::EclipseState state(deck);
+    Opm::EclipseGrid eclipse_grid = state.getInputGrid();
+
+    BOOST_CHECK_EQUAL( eclipse_grid.getTotalActiveLGR() , 25U );
+    BOOST_CHECK_EQUAL( eclipse_grid.lgr_children_cells[0].getTotalActiveLGR() , 9U );
+    BOOST_CHECK_EQUAL( eclipse_grid.lgr_children_cells[1].getTotalActiveLGR() , 9U );
+    
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("GLOBAL",0,0,0), 0U);
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("GLOBAL",2,2,0), 24U);
+
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("LGR1",0,0,0), 12U);
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("LGR1",2,2,0), 20U);
+
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("LGR2",0,0,0), 1U);
+    BOOST_CHECK_EQUAL(eclipse_grid.getActiveIndexLGR("LGR2",2,2,0), 9U);
```

Comment:
Could we add some checks to trigger the throws? For example, trying to request a non-existing local or global index from LGR, or a non-existing LGR label.